### ISSUE TITLE
Default Filter with Cookie extension Bug when clicking Clear button

### ISSUE THREAD

**Issue by mholtermann** [wenzhixin/bootstrap-table#6763]
### Bootstraptable version(s) affected

1.21.4

### Description

**Describe the bug**
Combining the Filter Control extension and cookie extension creates a bug when users click the clear button and have filter defaults set on the table.
Without the cookie plugin filter defaults clear correctly and display matching data on no filter. 
With the cookie plugin the filter defaults clear and display data with no filters but the filter text or what is selected changes back to the default if the last filter you changed was not a filter with a default. If the last filter you changed was a filter with a default it appears to be cleared because your focus goes back to that field even though in the background the default is still saved, since if you press clear again the data will be restricted to the filter default even though the filter text or dropdown is empty because of your focus.

### Example(s)

**To Reproduce**

Case 1

https://user-images.githubusercontent.com/22463764/234330251-fb69864d-b9ac-4471-953e-fc3d9b9e471b.mp4

1. Go to https://live.bootstrap-table.com/code/mholtermann/15120
2. Change the last filter to $0
3. Click clear, it looks ok but if you click clear again you will see that the filter is still holding the default even though the dropdown has blank selected.

Case 2

https://user-images.githubusercontent.com/22463764/234331325-ff5ae755-805b-471a-88ac-21a09ca1dd25.mp4

1. Go to https://live.bootstrap-table.com/code/mholtermann/15120
2. Change the last filter to $0
2. Change the middle filter to Ite
3. Click clear, the last dropdown says $0 is selected but the data does not match. If you click clear again the data will start to match the default filters again, the middle filter will still be blank even though the filter is passing the default in the backend.

Extra Example

https://user-images.githubusercontent.com/22463764/234331542-42366dd6-e7a3-4590-8e3f-0927219271f7.mp4

### Possible Solutions

Basically I think after hitting clear, the data should key off of the default filters, it only keys off of the default filters after pressing clear twice. Also when clicking clear all filters should be filled out with the default filter value even if your focus is given to the input or select element. After this fix probably a good idea would be to rename the button to "Reset" instead of clear, and then create new functionality for a clear button to wipe out the default filters when hitting clear.

**Comment by mholtermann** [comment]
Any one have any ideas of a work around? I thought maybe I could refresh the table to basically double click the button.
None of the below work.
`$("[name=\'clearSearch\']").click(function() {
				 //$(\'#maintenance_table_'.$table_id.'\').bootstrapTable(\'clearFilterControl\');
				 //$(\'#maintenance_table_'.$table_id.'\').bootstrapTable(\'resetSearch\');
				// $(\'#maintenance_table_'.$table_id.'\').bootstrapTable(\'triggerSearch\');
				 //$(\'#maintenance_table_'.$table_id.'\').bootstrapTable(\'refresh\').bootstrapTable(\'refreshOptions\');
				//$(\'#maintenance_table_'.$table_id.'\').bootstrapTable(\'clearFilterControl\');
});`

**Comment by mholtermann** [comment]
This is still an issue in 1.22.6 by changing the libraries in the example
